### PR TITLE
add getStaticProps map

### LINKED ISSUE #456: Use getStaticProps to render map cards

### PR DESCRIPTION
**Pull request by ada-only-one-ada**
Fixed #456

### CODE REVIEW

**Comment by ada-only-one-ada** [comment]
The problem is because not passing {...pageProps} in the _app.js file. 
Ref: https://stackoverflow.com/questions/65248299/nextjs-getstaticprops-not-passing-returned-props-to-page

File: src/pages/map/index.tsx [hunk 1]
```diff
@@ -14,39 +14,41 @@ import {
...
 import LandmarkDrawer from 'components/LandmarkDrawer';
+import path from 'path';
+import { promises as fs } from 'fs';
 
 const desktopTextLimit = 80;
 const mobileTextLimit = 30;
 
-export default function Page() {
+type PageProps = {
+  data: Record<string, Landmark[]>;
```
Comment (weihanwang) [inline src/pages/map/index.tsx:29]: Can you do `Record<Latitude, Landmark[]>`?

File: src/pages/map/index.tsx [hunk 2]
```diff
@@ -14,39 +14,41 @@ import {
...
+};
+
+const latitudes = ['个人成长', '事业发展', '社会责任'];
```
Comment (weihanwang) [inline src/pages/map/index.tsx:32]: use `Latitudes` defined in shared/Map.ts instead

File: src/pages/map/index.tsx [hunk 3]
```diff
@@ -14,39 +14,41 @@ import {
...
+export default function Page({ data }: PageProps) {
   const [selectedLandmark, setSelectedLandmark] = useState<Landmark | null>(null);
 
   return <>
     <TabsWithUrlParam isLazy>
       <TabList>
-        <Tab>个人成长</Tab>
-        <Tab>事业发展</Tab>
-        <Tab>社会责任</Tab>
+        {latitudes.map(latitude =>
```
Comment (weihanwang) [inline src/pages/map/index.tsx:40]: it should iterate over the keys of data and not latitudes.

File: src/pages/map/index.tsx [hunk 4]
```diff
@@ -14,39 +14,41 @@ import {
...
-        </TabPanel>
-
-        <TabPanel>
-          <LandmarkTabPanel latitude="事业发展" selectLandmark={setSelectedLandmark}/>
-        </TabPanel>
-
-        <TabPanel>
-          <LandmarkTabPanel latitude="社会责任" selectLandmark={setSelectedLandmark}/>
-        </TabPanel>
+        {latitudes.map(latitude =>
```
Comment (weihanwang) [inline src/pages/map/index.tsx:46]: ditto

File: src/pages/map/index.tsx [hunk 5]
```diff
@@ -57,18 +59,17 @@ export default function Page() {
...
-  latitude: Latitude; 
+const LandmarkTabPanel = ({ landmarks, selectLandmark }: { 
+  landmarks: Landmark[];
   selectLandmark: (landmark: Landmark) => void 
 }) => {
-  const { data, isLoading } = trpcNext.map.listLandmarks.useQuery(latitude);
   return (
     <>
-      {isLoading ? 
+      {!landmarks ?
```
Comment (weihanwang) [inline src/pages/map/index.tsx:68]: `!landmarks` should be never true. remove line 68 and 69.

File: src/pages/map/index.tsx [hunk 6]
```diff
@@ -14,39 +14,41 @@ import {
...
-          <LandmarkTabPanel latitude="事业发展" selectLandmark={setSelectedLandmark}/>
-        </TabPanel>
-
-        <TabPanel>
-          <LandmarkTabPanel latitude="社会责任" selectLandmark={setSelectedLandmark}/>
-        </TabPanel>
+        {latitudes.map(latitude => 
+          <TabPanel key={latitude}>
+            <LandmarkTabPanel landmarks={data[latitude]} 
+              selectLandmark={setSelectedLandmark}/>
```
Comment (weihanwang) [inline src/pages/map/index.tsx:49]: space before `/>`

File: src/pages/map/index.tsx [hunk 7]
```diff
@@ -97,3 +98,31 @@ const LandmarkCard = ({ landmark, selectLandmark }: {
     <CardBody>{cardText}</CardBody>
   </Card>; 
 };
+
+export async function getStaticProps(){
```
Comment (weihanwang) [inline src/pages/map/index.tsx:102]: space before `{`

File: src/pages/map/index.tsx [hunk 8]
```diff
@@ -97,3 +85,32 @@ const LandmarkCard = ({ landmark, selectLandmark }: {
     <CardBody>{cardText}</CardBody>
   </Card>; 
 };
+
+export async function getStaticProps() {
+  const latitudes = Latitudes;
```
Comment (weihanwang) [inline src/pages/map/index.tsx:90]: remove this variable. refer to `Latitudes` directly in code below.

**Review by weihanwang — APPROVED** [review]

**Pull request merged**